test/integration_tests/test_open.py

```diff
@@ -0,0 +1,76 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import unittest
+from test.util import create_sample_image, file_remove, load_inbuilt_udfs
+
+import numpy as np
+import pandas as pd
+
+from eva.catalog.catalog_manager import CatalogManager
+from eva.configuration.configuration_manager import ConfigurationManager
+from eva.models.storage.batch import Batch
+from eva.server.command_handler import execute_query_fetch_all
+from eva.storage.storage_engine import StorageEngine
+
+
+class OpenTests(unittest.TestCase):
+    def setUp(self):
+        CatalogManager().reset()
+        ConfigurationManager()
+
+        # Load built-in UDFs.
+        load_inbuilt_udfs()
+
+        # Insert image path.
+        create_sample_image()
+        create_table_query = "CREATE TABLE IF NOT EXISTS testOpenTable (num INTEGER);"
+        execute_query_fetch_all(create_table_query)
+
+        # Insert dummy data into table.
+        table_df_metadata = CatalogManager().get_dataset_metadata(None, "testOpenTable")
+        storage_engine = StorageEngine().factory(table_df_metadata)
+        storage_engine.write(
+            table_df_metadata, Batch(pd.DataFrame([{"num": 1}, {"num": 2}]))
+        )
+
+    def tearDown(self):
+        file_remove("dummy.jpg")
+
+        # Drop table.
+        drop_table_query = "DROP TABLE testOpenTable;"
+        execute_query_fetch_all(drop_table_query)
+
+    def test_open_should_open_image(self):
+        # Test query runs successfully with Open function call.
+        config = ConfigurationManager()
+        upload_dir_from_config = config.get_value("storage", "upload_dir")
+        img_path = os.path.join(upload_dir_from_config, "dummy.jpg")
+        select_query = """SELECT num, Open("{}") FROM testOpenTable;""".format(img_path)
+        batch_res = execute_query_fetch_all(select_query)
+
+        expected_img = np.array(np.ones((3, 3, 3)), dtype=np.uint8)
+        expected_img[0] -= 1
+        expected_img[2] += 1
+
+        num = batch_res.column_as_numpy_array("testopentable.num")[0]
+        self.assertEqual(1, num)
+        actual_img = batch_res.column_as_numpy_array("open.data")[0]
+        self.assertEqual(np.sum(expected_img), np.sum(actual_img))
+
+        num = batch_res.column_as_numpy_array("testopentable.num")[1]
+        self.assertEqual(2, num)
+        actual_img = batch_res.column_as_numpy_array("open.data")[1]
+        self.assertTrue(np.isnan(actual_img))
```

Comment:
Shouldn't the second row also have the same data?

Reply:
https://github.com/georgia-tech-db/eva/blob/4da3e482728ad157fbd0e9e2509fda1295ea1759/eva/expression/function_expression.py#L89

Reply:
Actually, during my testing, the second row never triggers another `Open` call. I thought it is by design.